repos/system_upgrade/common/actors/targetuserspacecreator/libraries/userspacegen.py

```diff
@@ -122,16 +123,38 @@ def _update_files(copy_files):
             raise StopActorExecutionError('No storage info available cannot proceed.')
 
 
+def _restore_persistent_package_cache(userspace_dir):
+    if get_env('LEAPP_DEVEL_USE_PERSISTENT_PACKAGE_CACHE', None) == '1':
+        if os.path.exists(PERSISTENT_PACKAGE_CACHE_DIR):
+            with mounting.NspawnActions(base_dir=userspace_dir) as target_context:
+                target_context.copytree_to(PERSISTENT_PACKAGE_CACHE_DIR, '/var/cache/dnf')
+    # We always want to remove the persistent cache here to unclutter the system
+    run(['rm', '-rf', PERSISTENT_PACKAGE_CACHE_DIR])
+
+
+def _backup_to_persistent_package_cache(userspace_dir):
+    if get_env('LEAPP_DEVEL_USE_PERSISTENT_PACKAGE_CACHE', None) == '1':
+        # Clean up any dead bodies, just in case
+        run(['rm', '-rf', PERSISTENT_PACKAGE_CACHE_DIR])
+        if os.path.exists(os.path.join(userspace_dir, 'var', 'cache', 'dnf')):
+            with mounting.NspawnActions(base_dir=userspace_dir) as target_context:
+                target_context.copytree_from('/var/cache/dnf', PERSISTENT_PACKAGE_CACHE_DIR)
+
+
 def prepare_target_userspace(context, userspace_dir, enabled_repos, packages):
     """
     Implement the creation of the target userspace.
     """
+    _backup_to_persistent_package_cache(userspace_dir)
```

Comment:
We actually want to enter the restore always, because of the removal of the left over. I am not sure how putting it outside would actually make it any more convenient.

Reply:
Not always but only when 'LEAPP_DEVEL_USE_PERSISTENT_PACKAGE_CACHE' is set. But anyway, let's ignore it.

Reply:
No, always to ensure the left over cache gets removed when the env variable isn't set

Reply:
Naah, I see it now...